pkg/machines/components/create-vm-dialog/pxe-helpers.js

```diff
@@ -0,0 +1,187 @@
+                return 'Routed Network';
+        } else {
+            if (dev)
+                return cockpit.format('$0 to $1', mode, dev);
+            else
+                return cockpit.format('$0 network', mode.toUpperCase());
+        }
+    } else {
+        return _("Isolated network");
+    }
+}
+
+/**
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {object} virtualNetwork - A Virtual Network object.
+ */
+export function getVirtualNetworkPXESupport(virtualNetwork) {
+    if (virtualNetwork.forward && virtualNetwork.forward.mode != 'nat') {
+        return true;
+    }
+
+    let ips = virtualNetwork.ip;
+
+    for (let i = 0; i < ips.length; i++)
+        if (ips[i].dhcp.bootp)
+            return true;
+
+    return false;
+}
+
+/**
+ * Returns the first available Network Resource to be used for showing to PXE Network Sources list.
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {array} virtualNetworks - An array of object containing Virtual Networks.
+ * @param {string} connectionName - The libvirt connection name.
+ */
+export function getPXEInitialNetworkSource(nodeDevices, virtualNetworks, connectionName) {
+    if (virtualNetworks.length > 0)
+        return cockpit.format('network=$0', virtualNetworks[0].name);
+
+    let netNodeDevices = getNodeDevicesOfType(nodeDevices, connectionName, 'net');
+
+    /* Do not show to the user the libvirt virbrX-nic devices since these are
+     * supposed to be managed through Virtual Networks
+     */
+    let libvirtVirBridges = getLibvirtNetworkBridges(virtualNetworks);
+    netNodeDevices = netNodeDevices.filter(netNodeDevice => {
+        for (let i in libvirtVirBridges) {
+            if (netNodeDevice.capability.interface == (libvirtVirBridges[i] + '-nic'))
+                return false;
+        }
+        return true;
+    });
+
+    if (netNodeDevices.length > 0)
+        return cockpit.format('type=direct,source=$0', netNodeDevices[0].capability.interface);
+}
+
+/**
+ * Returns the Select Entries rows for the PXE Network Sources.
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {array} virtualNetworks - An array of object containing Virtual Networks.
+ * @param {string} connectionName - The libvirt connection name.
+ */
+export function getPXENetworkRows(nodeDevices, virtualNetworks, connectionName) {
+    /* Do not show to the user the libvirt virbrX-nic devices since these are
+     * supposed to be managed through virtual networks
+     */
+    const libvirtVirBridges = getLibvirtNetworkBridges(virtualNetworks);
+    let netNodeDevices = getNodeDevicesOfType(nodeDevices, connectionName, 'net');
+
+    netNodeDevices = netNodeDevices.filter(netNodeDevice => {
+        for (let i in libvirtVirBridges) {
+            if (netNodeDevice.capability.interface == (libvirtVirBridges[i] + '-nic'))
+                return false;
+        }
+        return true;
+    });
```

Comment:
This is almost the same code as above, can you please factorize it out into a helper?

Reply:
done